### DISCUSSION TITLE
Mobile support

### DISCUSSION THREAD

**Discussion by moklick** [xyflow/xyflow#1677] [New Features]
add basic functionality for connecting nodes on touch devices

**Comment by utzy** [comment]
Would be great to see this!
The node editor is just not usable on touch devices at the moment, which is a deal breaker for our use case on iPad.
I have managed to get _either_ pan and zoom _OR_ connecting nodes to work...not both at the same time though.
Rete.js does this perfectly.

Is there a known workaround for this limitation?
If so, please let me know...otherwise we will have to go with Rete.js

**Comment by cybaj** [comment]
I hope to see this and also want to know any workaround for this, too. please let me know.

**Comment by Alexey174** [comment]
I really like React Flow. I've been doing desktop development for about six months. And all this time I am waiting for the support of the mobile version

**Comment by OleksandrCEO** [comment]
I am CEO and founder of a huge SaaS (for business automation). And I would like to use your script for my projects. We can even mention you on our site or help with developing. But mobile support is one of most important parts of our project!

You can change life for better of million clients of thousands our customers. Is this is not important reason to change priority of this task?

**Reply by moklick** [reply]
Hey @skyjoom. You could help us by [sponsoring us](https://github.com/sponsors/wbkd) or open a PR. Right now we are working on subgraphs and a better state management. It is on our list but not as highly prioritized as the other tasks i just mentioned.

**Comment by leejustin** [comment]
> I am CEO and founder of a huge SaaS (for business automation). And I would like to use your script for my projects. We can even mention you on our site or help with developing. But mobile support is one of most important parts of our project!
> 
> You can change life for better of million clients of thousands our customers. Is this is not important reason to change priority of this task?

I hope this doesn't come off as hostile since that's not my intention but if you really are a CEO of such a large company, perhaps you could be more supportive of @moklick by becoming a sponsor for this project. Or, you could have someone on your team begin to help look at mobile support or some other easier tasks and contribute back to this project. This project has a lot of traction and simply offering to mention the project for visibility does not seem very compelling.

He is incredibly busy yet is able to offer all of us this library for free. The code is incredibly high quality for an open source project.

**Comment by moklick** [comment]
Basic support for touch screens will be available in the next major release. You can find the new features, changes and installation instructions for the beta in this PR #1555 if you want to test it.
